### PR TITLE
Add support for PCRE2

### PR DESCRIPTION
**Pull request by csware**
Fixes issue #40.

### CODE REVIEW

**Comment by csware** [comment]
I updated the PR.

I kept PCRE in the initial PR as maintaining both did not look too complicated and I thought that maybe some people still use PCRE and not PCRE2.

**Comment by xuhdev** [comment]
I think you can designate Ubuntu 16.04 as the build environment: https://docs.travis-ci.com/user/reference/xenial/#using-xenial

In the meanwhile, can you rebase the PR? Thanks!

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,14 @@ compiler:
   - gcc
   - clang
 
+dist: xenial
+
+addons:
+  apt:
+    packages:
+      - cmake
+     - libpcre2-dev
```
[the remark this answers is not in the document]
Comment (csware) [reply .travis.yml]: I would expect to get at least an error...

[the remark this answers is not in the document]
Comment (csware) [reply .travis.yml]: Seems so

**Comment by xuhdev** [comment]
How about [this solution](https://stackoverflow.com/a/29871891/1150462)? I'm OK with bumping minimum required cmake version to 3.5.1 (the one in Ubuntu 16.04). Debian Jessie users should also be fine as 3.6.2 has been backported for them. I would rather drop the support of Ubuntu 14.04 in favor of easier maintenance.

**Pull request merged**